examples/golang-network/envoy.yaml

```diff
@@ -0,0 +1,29 @@
+# envoy demo with golang extension enabled
+static_resources:
+  listeners:
+  - name: listener_0
+    address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 10000
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.golang
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.golang.v3alpha.Config
+          is_terminal_filter: true
+          library_id: simple
+          library_path: "/lib/simple.so"
+          plugin_name: simple
+          plugin_config:
+            "@type": type.googleapis.com/xds.type.v3.TypedStruct
+            value:
+              echo_server_addr: echo_service
+  clusters:
+  - name: plainText
+    type: ORIGINAL_DST
+    lb_policy: CLUSTER_PROVIDED
+    connect_timeout: 10s
+    original_dst_lb_config:
```

Comment:
this is required since we use it to inform the origin dst cluster to create upstream connection with target ip. but i think we could achieve this with a better approach later on.

Reply:
if there is a better/simpler way lets try and do that (not entirely clear i understand why its needed)

for these sandboxes we want the simplest, minimal config/setup we can, otherwise the end user is not sure what is/not required and will copy out any superfluous configs etc

Reply:
ok, i see - this is required to use the configured upstream target

Reply:
the thing that is making my uncomfortable about this is that an L4 filter has

```yaml
use_http_header: true
```

configured - which seems pretty counterintuitive

Reply:
yes, it's quite counterintuitive, I open up an issue https://github.com/envoyproxy/envoy/issues/28300 for this. I will investigate better way for the upstream host selection.